app/models/task.rb

```diff
@@ -47,6 +48,10 @@ def pending?
     !complete?
   end
 
+  def expired?
+    self.due_date && (self.due_date < Time.now && !complete?)
```

Comment:
nitpick `self.due_date && (self.due_date < Time.now && pending?)`
it's a lot easier to read/understand true-positive conditions